packages/components/src/dropdown-menu/README.md

```diff
@@ -131,80 +131,72 @@ const MyDropdownMenu = () => (
 
 The component accepts the following props:
 
-#### icon
+#### `icon`: `string | null`
 
 The [Dashicon](https://developer.wordpress.org/resource/dashicons/) icon slug to be shown in the collapsed menu button.
 
--   Type: `String|null`
 -   Required: No
 -   Default: `"menu"`
 
 See also: [https://developer.wordpress.org/resource/dashicons/](https://developer.wordpress.org/resource/dashicons/)
 
-#### label
+#### `label`: `string`
 
 A human-readable label to present as accessibility text on the focused collapsed menu button.
 
--   Type: `String`
 -   Required: Yes
 
-#### controls
+#### `controls:` `DropdownOption[]`
 
 An array of objects describing the options to be shown in the expanded menu.
 
 Each object should include an `icon` [Dashicon](https://developer.wordpress.org/resource/dashicons/) slug string, a human-readable `title` string, `isDisabled` boolean flag and an `onClick` function callback to invoke when the option is selected.
 
-A valid DropdownMenu must specify one or the other of a `controls` or `children` prop.
-
--   Type: `Array`
+A valid DropdownMenu must specify a `controls` or `children` prop, or both.
 -   Required: No
 
-#### children
+#### `children`: `( callbackProps: DropdownCallbackProps ) => ReactNode`
 
 A [function render prop](https://reactjs.org/docs/render-props.html#using-props-other-than-render) which should return an element or elements valid for use in a DropdownMenu: `MenuItem`, `MenuItemsChoice`, or `MenuGroup`. Its first argument is a props object including the same values as given to a [`Dropdown`'s `renderContent`](/packages/components/src/dropdown#rendercontent) (`isOpen`, `onToggle`, `onClose`).
 
-A valid DropdownMenu must specify one or the other of a `controls` or `children` prop.
+A valid DropdownMenu must specify a `controls` or `children` prop, or both.
 
--   Type: `Function`
 -   Required: No
 
 See also: [https://developer.wordpress.org/resource/dashicons/](https://developer.wordpress.org/resource/dashicons/)
 
-#### className
+#### `className`: `string`
 
 A class name to apply to the dropdown menu's toggle element wrapper.
 
--   Type: `String`
 -   Required: No
 
-#### popoverProps
+#### `popoverProps`: `DropdownProps[ 'popoverProps' ]`
 
 Properties of `popoverProps` object will be passed as props to the nested `Popover` component.
 Use this object to modify props available for the `Popover` component that are not already exposed in the `DropdownMenu` component, e.g.: the direction in which the popover should open relative to its parent node set with `position` prop.
 
--   Type: `Object`
 -   Required: No
 
-#### toggleProps
+<!-- NTS: This is vague, but the actual type notation feels like a bit much for the README. -->
+#### `toggleProps`: `Object`
```

Comment:
This is vague, but listing the type as `ToggleProps` didn't feel helpful, and notating the full type felt like it was a bit much for the README.